### ISSUE TITLE
Proposal: Human readable/sortable version identifier

### ISSUE THREAD

**Issue by tyler569** [ziglang/zig#2343]
Currently when you go to the Zig website and go to download the master branch, it is identified by the version number (bumped once every 6 months) and a commit hash, which is unpredictable and changes several times a day often.

Many projects add an additional field to their version identifier that indicates the number of commits since the last tag, giving a human-readable and sortable ordering to version numbers between tags.  As an example of how this looks, taking the `git describe` default format in the Zig repository as it is now:
```
$ git describe
0.4.0-82-gd44d2784
```

The main benefit as I see it is that when you want to update your Zig compiler, you can very easily tell just how far behind you are, and you can sort between multiple downloads in a downloads folder without having to remember commit hashes.

Basically, what would you rather see 😃?
```
$ ls Downloads
zig-linux-x86_64-0.4.0+73421833
zig-linux-x86_64-0.4.0+a66b8729
zig-linux-x86_64-0.4.0+f25393ee
zig-linux-x86_64-0.4.0+0f8fc3b9
```
or
```
$ ls Downloads
zig-linux-x86_64-0.4.0+12-g73421833
zig-linux-x86_64-0.4.0+44-ga66b8729
zig-linux-x86_64-0.4.0+67-gf25393ee
zig-linux-x86_64-0.4.0+82-g0f8fc3b9
```

**Comment by mikdusan** [comment]
> Basically, what would you rather see 😃?

this because my brain doesn't grok base-17 right away

>```
> $ ls Downloads
> zig-linux-x86_64-0.4.0+12-73421833
> zig-linux-x86_64-0.4.0+44-a66b8729
> zig-linux-x86_64-0.4.0+67-f25393ee
> zig-linux-x86_64-0.4.0+82-0f8fc3b9
> ```

**Comment by tyler569** [comment]
Ah, should have removed that - git puts the 'g' the for compatibility with other VCS's (if you pulled a release from SVN you could use -r(rev number) for example), but we obviously wouldn't need that.

**Comment by andrewrk** [comment]
I think status quo is good enough. I'd like to focus on other things.
